src/panorama/scripts/custom_game/game_setup.js

```diff
@@ -2686,52 +2687,52 @@ function OnSkillTabShown(tabName) {
                     if(filterInfo.shouldShow) {
                         if(useSmartGrouping) {
                             var theOwner = abilityHeroOwner[abilityName];
+                            var neutralGroup = flagDataInverse[abilityName].group;
 
-                            if(theOwner != null) {
+                            if(theOwner != null || neutralGroup != null) {
                                 // Group it
-                                var groupCon = heroOwnerBlocks[theOwner];
+                                var groupKey = theOwner != null ? theOwner : neutralGroup;
+                                var groupCon = groupBlocks[groupKey];
                                 if(groupCon == null) {
-                                    groupCon = $.CreatePanel('Panel', con, 'group_container_' + theOwner);
+                                    groupCon = $.CreatePanel('Panel', con, 'group_container_' + groupKey);
                                     groupCon.SetHasClass('grouped_skills', true);
-
-                                    // Store it
-                                    heroOwnerBlocks[theOwner] = groupCon;
                                 }
 
+                                groupBlocks[groupKey] = groupCon;
+
+                                toSort.push({
+                                    txt: groupKey,
+                                    con: groupCon,
+                                    grouped: true
+                                });
+
                                 // Making the layout much nicer
-                                if(heroBlockCounts[theOwner] == null) {
-                                    heroBlockCounts[theOwner] = 1;
+                                if(blockCounts[groupKey] == null) {
+                                    blockCounts[groupKey] = 1;
                                 } else {
-                                    ++heroBlockCounts[theOwner];
+                                    ++blockCounts[groupKey];
 
-                                    if(heroBlockCounts[theOwner] == 3) {
+                                    if(blockCounts[groupKey] == 3) {
```

Comment:
You can probably pull this out from the if/else. Nested branching is hard to follow.